samples/CustomMiddleware/MyCustomMiddleware.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Middleware;
+using Microsoft.Extensions.Logging;
+
+namespace CustomMiddleware
+{
+    public class MyCustomMiddleware : IFunctionsWorkerMiddleware
+    {
+        public async Task Invoke(FunctionContext context, FunctionExecutionDelegate next)
```

Comment:
Would be nice if instead of a Delegate it would be `Func<FunctionContext, Task>`

Reply:
That’s essentially what this delegate is, but a specialized version specific to this context, which is helpful in a number of scenarios.

Reply:
This delegate is also part of the public API. I do have plans to introduce UseMiddleware APIs that will take a simple lambda (or a `Func` with the appropriate signature), but this is the basic delegate that ends up being used and is safe to use in some contexts where it would otherwise become ambiguous without extra hoops.

Can you expand on concerns you had with this? Is this mostly about the usability? And if so, does the plan to support the APIs I mentioned above help with that?